fibonacci.js

```diff
@@ -0,0 +1,158 @@
+  finalResult.innerText = fSequence[fSequence.length - 1];
+}
+
+let button = document.getElementById("fButton");
+let input = document.getElementById("fInput");
+let finalResult = document.getElementById("displayF");
+let error = document.querySelector(".error");
+let serverError = document.getElementById("serverError");
+let resultContainer = document.querySelector(".resultContainer");
+let spinnerResult = document.getElementById("spinnerResult");
+let saveCalculation = document.getElementById("saveCalculation");
+let select = document.getElementById("select")
+let showSelectedValue = document.querySelector(".text-muted")
+
+button.addEventListener("click", chooseLocalOrServer);
+input.addEventListener("focus", clearInput);
+
+function chooseLocalOrServer() {
+    if (input.value.length) {
+        button.disabled = true;
+   
+    if (input.value > 50) {
+        input.classList.add("border-danger", "text-danger");
+        error.classList.remove("d-none");
+      } 
+  else if (saveCalculation.checked) {
+    runServerFibonacci();
+  } else  if (input.value<=0){
+    finalResult.classList.remove(
+        "spinner-border",
+        "text-primary",
+        "font-weight-bold"
+      );
+      finalResult.classList.add("text-danger");
+      finalResult.innerText = "Error: number can't be smaller than 1";
+    } else {
+    runLocalFibonacci(input.value);
+  }
+}}
+
+function runServerFibonacci() {
+      finalResult.classList.add("spinner-border", "text-primary");
+      fetch(`http://localhost:5050/fibonacci/${input.value}`)
+        .then((res) => {
+          if (res.ok) return res.json();
+          else {
+            return res.text().then((text) => {
+              throw new Error(text);
+            });
+          }
+        })
+        .then((data) => {
+          finalResult.classList.remove(
+            "spinner-border",
+            "text-primary",
+            "font-weight-bold"
+          );
+          finalResult.classList.add("fs-4", "text-decoration-underline");
+          finalResult.innerText = data.result;
+          fetchResults(selectedValue);
+        })
+        .catch((err) => {
+          finalResult.classList.remove(
+            "spinner-border",
+            "text-primary",
+            "font-weight-bold"
+          );
+          finalResult.classList.add("text-danger");
+          finalResult.innerText = err;
+        });
+    
+  }
+
+
+function clearInput() {
+  input.value = " ";
+  finalResult.innerText = " ";
+  error.classList.add("dNone");
+  input.classList.remove("border-danger", "text-danger", "fs-6");
+  serverError.innerText = " ";
+  finalResult.classList.remove(
+    "text-danger",
+    "fs-4",
+    "text-decoration-underline"
+  );
+  error.classList.add("d-none");
+  button.disabled = false;
+}
+
+function fetchResults(selectedValue) {
+  spinnerResult.classList.remove("d-none");
+  fetch("http://localhost:5050/getFibonacciResults")
+    .then((response) => response.json())
+    .then((data) => {
```

Comment:
When you use .then(), make sure you always have .catch() block in order to handle possible errors.